pkg/updater/download.go

```diff
@@ -116,7 +116,11 @@ func extractTarGz(archivePath string, destinationPath string) error {
 		if err != nil {
 			return fmt.Errorf("could not read tar header: %w", err)
 		}
-		target := filepath.Join(destinationPath, header.Name)
+		cleanName := filepath.Clean(header.Name)
+		if strings.HasPrefix(cleanName, "..") {
+			return fmt.Errorf("tar entry %s is trying to escape the destination directory", header.Name)
+		}
+		target := filepath.Join(destinationPath, cleanName)
```

Comment:
@Zenithar I'd assume that we want to preferably check if the abs path of the joined path is outside of the target dir since the `..` escape check may not be adequate but I could be wrong

Reply:
I switched to absolute comparison to remove any doubts.